instrumentation/cassandra/cassandra-4.4/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_4/CassandraAttributesExtractor.java

```diff
@@ -36,6 +38,12 @@ public void onEnd(
 
     Node coordinator = executionInfo.getCoordinator();
     if (coordinator != null) {
+      SocketAddress address = coordinator.getEndPoint().resolve();
```

Comment:
Should be fine for https://github.com/apache/cassandra-java-driver/blob/8d5849cb38995b312f29314d18256c0c3e94cf07/core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultEndPoint.java#L40 but not so fine for https://github.com/apache/cassandra-java-driver/blob/8d5849cb38995b312f29314d18256c0c3e94cf07/core/src/main/java/com/datastax/oss/driver/internal/core/metadata/SniEndPoint.java#L56

Reply:
oh, good catch! looks like we had that issue previously:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/16adb1d87bf596b35ea54577370ffe269f856188/instrumentation/cassandra/cassandra-4.4/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_4/CassandraNetworkAttributesGetter.java#L29-L31

it looks like we could use `getServerName()` if it's an `SniEndPoint`(?)

Reply:
what about port?

Reply:
i'm not sure, but `getServerName()` may include port